tools/add_triage_signature.py

```diff
@@ -455,12 +542,57 @@ def get_jira_client(username, password):
     logger.info("log-in with username: %s", username)
     return jira.JIRA(JIRA_SERVER, basic_auth=(username, password))
 
+
 ############################
 # Signature runner functionality
 ############################
 LOGS_URL_FROM_DESCRIPTION_OLD = re.compile(r".*logs:\* \[(http.*)\]")
 LOGS_URL_FROM_DESCRIPTION_NEW = re.compile(r".*\* \[[lL]ogs\|(http.*)\]")
 
+
+def search_strings_in_file(file_name: str, substr: List[str]) -> List[str]:
+    with open(file_name, "r") as _file:
+        data = _file.read()
+
+    res = []
+
+    for string in substr:
+        regex = re.compile(r'.*%s.*' % string)
+        res += regex.findall(data)
+
+    return res
+
+
+def group_similar_strings(ls: List[str], ratio: int) -> List[str]:
+    groups = list()
+    for name in ls:
+        for group in groups:
+            if all(fuzz.ratio(name, w) > ratio for w in group):
+                group.append(name)
+                break
+        else:
+            groups.append([name, ])
+
+    return groups
+
+
+def get_host_id_for_host_tar(tar_path: str) -> str:
+    with TemporaryDirectory() as tempdir:
+        prefix = "logs_host_"
+        file_name = extract_file_from_tar(tar_path, prefix, tempdir)
+        return file_name[file_name.find(prefix):].strip(prefix)
+
+
+def extract_file_from_tar(tar_path: str, file_prefix: str, output_dir: str) -> str:
+    with tarfile.open(tar_path) as tar:
+        for member in tar.getnames():
+            if file_prefix in member:
```

Comment:
so it's not really prefix, right?
maybe pattern?

Reply:
I changed into `file_substring`